app/models/nonprofit.rb

```diff
@@ -277,4 +278,10 @@ def self.create_cache_key_for_id(id)
   def self.create_cache_key_for_location(state_code,city, name)
     "nonprofit__CACHE_KEY__LOCATION___#{state_code}____#{city}___#{name}"
   end
+
+  private
+
+  def timezone_is_valid
+    timezone.blank? || ActiveSupport::TimeZone.all.map{ |t| t.tzinfo.name }.include?(timezone) || errors.add(:timezone, 'is not a valid timezone')
+  end
```

Comment:
Since you mentioned that we have some invalid timezones in the DB, this validation will lead to failures when NPs try to save settings and haven't changed their current timezone. We'll have to correct the bad timezones in the DB.

Reply:
I suspect I know how these "invalid" timezones may have gotten added. If we were geocoding the nonprofits at some point to get their timezone, the geocoder probably didn't restrict the limited list of timezones in `ActiveSupport::TimeZone.all`. I think we should stick with `ActiveSupport::TimeZone.all` for now though and clean up the non-matching timezones in the DB.

Reply:
Considering that the list that users can choose from the settings page is fed by `ActiveSupport::TimeZone.all` I think we won't have problems with new timezones being created that are not compatible with the validation condition.

Reply:
I'll include in the script the fixes for those timezones that are not supported by `ActiveSupport::TimeZone.all`.